hancock/s3_utils.py

```diff
@@ -19,26 +19,32 @@ def client_options(cls, **kwargs):
     def generate_presigned_url(cls, Bucket, Key, Expiration=900):
 
         try:
-              s3_client = boto3.client('s3', **cls.client_options())
+              s3 = boto3.resource('s3', **cls.client_options())
 
         except ClientError as e:
              app.logger.debug(e)
              return {'presigned_url': None}, 404
 
         # check bucket and object exist
         try:
-             s3_client.head_object(Bucket=Bucket, Key=Key)
+
+            bucket = s3.Bucket(Bucket)
+            objs = bucket.objects.filter(Prefix=Key)
+
         except ClientError as e:
              app.logger.debug(e)
              return {'presigned_url': None}, 404
 
         try:
-            url = s3_client.generate_presigned_url('get_object',
-                                                        Params={'Bucket': Bucket,
-                                                                'Key': Key},
-                                                        ExpiresIn=Expiration)
+            url_ls =[]
+            for obj in objs.all():
+                app.logger.info(obj.key)
+                url_ls.append(s3.meta.client.generate_presigned_url('get_object',
+                                                            Params={'Bucket': Bucket,
+                                                                    'Key': obj.key},
+                                                            ExpiresIn=Expiration))
         except ClientError as e:
                  app.logger.debug(e)
                  return {'presigned_url': None}, 404
 
-        return {'presigned_url': url}, 200
\ No newline at end of file
+        return {'presigned_url': url_ls}, 200
```

Comment:
Should this be `jsonify`'ed as well?

Reply:
I have tried doing this and it causes a lot of errors as it is used as a sub function in routes. It probably does not need a 200 reponse as the response is handled by the @api.marshal_with decorator